v2/yarpcthrift/thriftrw-plugin-yarpc2/server.go

```diff
@@ -0,0 +1,132 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"path/filepath"
+
+	"go.uber.org/thriftrw/plugin"
+)
+
+const serverTemplate = `
+// Code generated by thriftrw-plugin-yarpc
+// @generated
+
+<$pkgname := printf "%sserver" (lower .Name)>
+package <$pkgname>
+
+<$yarpc     := import "go.uber.org/yarpc/v2">
+<$yarpcthrift    := import "go.uber.org/yarpc/v2/yarpcthrift">
+
+<$contextImportPath   := .ContextImportPath>
+<$unaryWrapperImport  := .UnaryWrapperImport>
+<$unaryWrapperFunc    := .UnaryWrapperFunc>
+
+</* Note that we import things like "context" inside loops rather than at the
+    top-level because they will end up unused if the service does not have any
+    functions.
+ */>
+
+// Interface is the server-side interface for the <.Name> service.
+type Interface interface {
+	<if .Parent><import .ParentServerPackagePath>.Interface
+	<end>
+	<range .Functions><if not .OneWay>
+		<$context := import $contextImportPath>
+		<.Name>(
+			ctx <$context>.Context, <range .Arguments>
+			<.Name> <formatType .Type>,<end>
+		)<if .ReturnType> (<formatType .ReturnType>, error)
+		<else> error
+		<end>
+	<end><end>
+}
+
+<$module := .Module>
+
+// New prepares an implementation of the <.Name> service for
+// registration.
+//
+// 	handler := <.Name>Handler{}
+// 	dispatcher.Register(<$pkgname>.New(handler))
+func New(impl Interface, opts ...<$yarpcthrift>.RegisterOption) []<$yarpc>.Procedure {
+	<if .Functions>h := handler{impl}<end>
+	service := <$yarpcthrift>.Service{
+		Name: "<.Name>",
+		Methods: []<$yarpcthrift>.Method{
+		<range .Functions><if not .OneWay>
+			<$yarpcthrift>.Method{
+				Name: "<.ThriftName>",
+				Handler: <import $unaryWrapperImport>.<$unaryWrapperFunc>(h.<.Name>),
+				Signature: "<.Name>(<range $i, $v := .Arguments><if ne $i 0>, <end><.Name> <formatType .Type><end>)<if .ReturnType> (<formatType .ReturnType>)<end>",
+				ThriftModule: <import $module.ImportPath>.ThriftModule,
+				},
+		<end><end>},
+	}
+
+	procedures := make([]<$yarpc>.Procedure, 0, <len .Functions>)
+	<if .Parent> procedures = append(procedures, <import .ParentServerPackagePath>.New(impl, opts...)...)
+	<end>         procedures = append(procedures, <$yarpcthrift>.BuildProcedures(service, opts...)...)
+	return procedures
```

Comment:
It looks like the changes introduced in https://github.com/yarpc/yarpc-go/pull/1482/files weren't ported into the plugin here. Was that intentional / planned for a later update? The original `spring` was likely branched off a version of `dev` before this change was introduced, so I suspect that it simply went unnoticed.

If we'd rather handle the porting later, please make a note in `CHANGES`. Some of the tests with regard to "A read/write client calling a read-only server" may break for the same reasons introduced in the PR I linked.

Reply:
You're right in your suspicion. Let's do the port after this lands. We'll need to do a pass of all changes merged to dev that we need to port back to spring, and this will be included.

Reply:
Yep, completely overlooked this. I'm going to put this on `CHANGES` and handle all the work done on dev after branching off later.